### PR TITLE
Feature/holidays

### PR DESCRIPTION
**Pull request by yashnerkar**

### CODE REVIEW

File: pages/staff/login/index.tsx [hunk 1]
```diff
@@ -76,16 +108,16 @@ function Login() {
...
-    if (!hiddenFeature) {
-      router.push("/doctor");
-    } else {
-      router.push("/").then(() => {
-        window.location.reload();
-      });
-    }
-  }, []);
+  // useEffect(() => {
+  //   const hiddenFeature = process.env.HIDDEN_FEATURE;
```
Comment (Mehul2410) [inline pages/staff/login/index.tsx:112]: remove comments

File: src/components/Admin/AppointmentTracker/AppointmentCard.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
+import React, { useEffect } from "react";
+import styles from "../../../../styles/Appointment.module.scss";
+import { useRouter } from "next/router";
+// import { updateAttendanceOfUser } from "../../utils/api/appointment";
```
Comment (Mehul2410) [inline src/components/Admin/AppointmentTracker/AppointmentCard.tsx:4]: remove comment

File: src/components/Admin/AppointmentTracker/AppointmentCard.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+              style={{
+                marginTop: "10px",
+              }}
+            >
+              <strong>Appointment Time: </strong> {appointment?.time}
+            </p>
+            {/* <Link
+              href={appointment?.meetingLink}
+              rel="noopener noreferrer"
+              target="_blank"
```
Comment (Mehul2410) [inline src/components/Admin/AppointmentTracker/AppointmentCard.tsx:127]: remove

File: src/components/Admin/CasePaperTracker/index.tsx
Comment (Mehul2410) [inline src/components/Admin/CasePaperTracker/index.tsx:1]: remove empty file

File: src/components/Admin/PrescriptionTracker/PrescriptionCard.tsx [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+  const generatePDF = () => {
+    const report = new jsPDF("portrait", "pt", "a1");
+    report.html(prescriptionRef.current, {
+      callback: function (pdf) {
+        pdf.save("prescription.pdf");
+      },
+    });
+  };
+
+  const style = {
```
Comment (Mehul2410) [inline src/components/Admin/PrescriptionTracker/PrescriptionCard.tsx:67]: next time create a folder and add these styles there

File: src/components/DashboardCards/index.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+type Props = {
+  text: string;
+  totalCount: number;
+};
+
+const DashboardCards = ({ totalCount, text }: Props) => {
+  return (
+    <div>
+      <div
+        style={{
```
Comment (Mehul2410) [inline src/components/DashboardCards/index.tsx:12]: use csss here

File: src/components/DownloadComponent/index.tsx [hunk 6]
```diff
@@ -11,7 +11,7 @@ type Props = {
 export default function DownloadsComponent({ downloadOptions }: Props) {
   return (
     <div className={styles.root}>
-      <h3>Downloads</h3>
+      {/* <h3>Downloads</h3> */}
```
Comment (Mehul2410) [inline src/components/DownloadComponent/index.tsx:14]: remove

File: src/components/Prescription/PrescriptionTable.tsx [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+interface Column {
+  id: string;
+  label: string;
+  minWidth?: number;
+  align?: "right" | "left" | "center";
+  format?: (value: number) => string;
+}
+
+const columns: readonly Column[] = [
+  // { id: "id", label: "Id", minWidth: 50 },
```
Comment (Mehul2410) [inline src/components/Prescription/PrescriptionTable.tsx:20]: remove comments

File: src/components/Prescription/index.tsx [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+
+      setDoctorAvailability(result);
+    };
+    if (doctor?.id) {
+      fetchData();
+    }
+  }, [doctor?.id]);
+
+  // const generatePDF = async () => {
+  //   const pdf = new jsPDF("portrait", "pt", "a4");
```
Comment (Mehul2410) [inline src/components/Prescription/index.tsx:64]: remove

File: src/components/holiday/HolidayModal.tsx
Comment (Mehul2410) [inline src/components/holiday/HolidayModal.tsx:1]: pls create a reusable component i feel the code is repeated

File: src/components/layout/Navigation.tsx [hunk 9]
```diff
@@ -1,18 +1,50 @@
...
-      {/* <div onClick={() => router.push('/chats')} className='navigationChat'>
-        <ChatIcon color={'#fff'} />
-      </div> */}
+      <div
+        style={{
+          flexGrow: 1,
+        }}
+      >
+        <img src="/logoWhite.png" style={{ height: "36px" }} />
+      </div>{" "}
```
Comment (Mehul2410) [inline src/components/layout/Navigation.tsx:35]: remove double quotes

File: src/components/logout/Logout.tsx [hunk 10]
```diff
@@ -1,47 +1,71 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { useRouter } from "next/router";
+// import {}
 import mixpanel from "mixpanel-browser";
```
Comment (Mehul2410) [inline src/components/logout/Logout.tsx:4]: remove

File: src/features/telehealth/TelehealthTabs.tsx
Comment (Mehul2410) [inline src/features/telehealth/TelehealthTabs.tsx:1]: remove it

File: src/theme/global.scss [hunk 11]
```diff
@@ -2,6 +2,8 @@
 @import url("https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,400;0,500;0,600;0,700;0,800;0,900;1,400;1,500;1,600;1,700;1,800&display=swap");
 @import url("https://fonts.googleapis.com/css2?family=Dancing+Script:wght@400;500;600;700&display=swap");
 
+$med: 768px;
```
Comment (Mehul2410) [inline src/theme/global.scss:5]: what is this?

File: src/theme/global.scss [hunk 12]
```diff
@@ -1033,3 +1048,4 @@ div[role="dialog"] {
   justify-content: center;
   align-items: center;
 }
+// max-width: 768px;
```
Comment (Mehul2410) [inline src/theme/global.scss:1051]: remove

File: src/theme/global.scss [hunk 13]
```diff
@@ -931,10 +934,21 @@ div[role="dialog"] {
   z-index: 10000;
 }
 
+.navigationBar {
+  @media screen and (max-width: 768px) {
+    max-width: 768px !important;
+    //do Smth
```
Comment (Mehul2410) [inline src/theme/global.scss:940]: what is this?

File: src/utils/api/index.tsx
Comment (Mehul2410) [inline src/utils/api/index.tsx:1]: remove unnecessary comments

**Review by Mehul2410 — CHANGES_REQUESTED** [review]

File: pages/admin/chats/index.tsx [hunk 14]
```diff
@@ -99,7 +120,7 @@ export default function Chats() {
           data: channels,
           error,
         }: PostgrestSingleResponse<privateChatType[]> = await supabase
-          .from("distinct_private_chats")
+          .from("distinct_private_chats4")
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/chats/index.tsx:123]: Its a 4

File: pages/admin/chats/private-chat/[id].tsx [hunk 15]
```diff
@@ -15,15 +18,27 @@ function PrivateChat() {
...
+    }
+  }, []);
 
   const sendMessageFunction = async () => {
     if (messageTyped.trim().length > 0) {
       try {
         const { data } = await supabase
           .from("private_chat")
+          // admin side
           .insert([
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/chats/private-chat/[id].tsx:42]: okay! I will keep in mind.
removed comment

File: pages/admin/chats/index.tsx [hunk 16]
```diff
@@ -88,7 +108,8 @@ export default function Chats() {
         )
         .subscribe();
     };
-
+    // if (token) {
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/chats/index.tsx:111]: removed comment

File: pages/admin/chats/index.tsx [hunk 17]
```diff
@@ -248,6 +271,7 @@ function PrivateChatGroupCard({
           style={{ justifyContent: "space-between", width: "100%" }}
         >
           <p className={styles.latestMessage}>
+            {/* single object */}
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/chats/index.tsx:274]: removed the comment

File: pages/admin/chats/index.tsx [hunk 18]
```diff
@@ -1,10 +1,13 @@
...
-import styles from "./../../../styles/Chats.module.scss";
+import styles from "../../../styles/AdminChats.module.scss";
 import { adminPhoneNumber } from "./private-chat/[id]";
 import { PostgrestSingleResponse } from "@supabase/supabase-js";
-
+import SideBar from "../../../src/components/Sidebar/Adminsidebar";
+import { UserPropsContext } from "../../../src/utils/context/user";
+import { adminRoles } from "..";
+import Swal from "sweetalert2";
 const receiverAdminPhone = "9594702015";
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/chats/index.tsx:11]: added to the env

File: pages/admin/doctor/registration/RegisterModal.tsx [hunk 19]
```diff
@@ -0,0 +1,55 @@
+import * as React from "react";
+import Box from "@mui/material/Box";
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/doctor/registration/RegisterModal.tsx:2]: sure! will implement this from now

File: pages/admin/doctor/registration/RegisterModal.tsx [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import Modal from "@mui/material/Modal";
+
+// Import the updated RegisterDoctor component
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/doctor/registration/RegisterModal.tsx:7]: removed the comment

File: pages/admin/holidays/AllHolidays.tsx [hunk 21]
```diff
@@ -0,0 +1,238 @@
+import React, { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import BasicTable from "../../../src/components/Tables/BasicTable";
+import moment from "moment";
+import { UserPropsContext } from "../../../src/utils/context/user";
+import HolidayModal from "../../../src/components/holiday/HolidayModal";
+// import DownloadsComponent from "../../components/DownloadComponent";
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/holidays/AllHolidays.tsx:7]: removed the comment

File: pages/admin/holidays/index.tsx [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+import Navigation from "../../../src/components/layout/Navigation";
+import Swal from "sweetalert2";
+import { adminRoles } from "..";
+type Props = {};
+
+const index = (props: Props) => {
+  const [open, setOpen] = useState(false);
+  const [holidays, setHolidays] = useState<any[]>([]);
+  const [holidaysCount, setHolidaysCount] = useState<number>(0);
+  // const { token } = React.useContext(UserPropsContext);
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/holidays/index.tsx:18]: removed the comment

File: pages/admin/holidays/index.tsx [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+  const handleOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const fetchAllHolidays = async (doctorId: number) => {
+    const result = await getAllHolidaysOfAllDoctors();
+    console.log(result);
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/holidays/index.tsx:30]: removed logs

File: pages/admin/users.tsx [hunk 24]
```diff
@@ -1,12 +1,35 @@
...
+import Navigation from "../../src/components/layout/Navigation";
+import { UserPropsContext } from "../../src/utils/context/user";
+import { useRouter } from "next/router";
+import { adminRoles } from "../../pages/admin";
+import Swal from "sweetalert2";
 type Props = {};
 
 function AdminHomePage({}: Props) {
+  const router = useRouter();
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/users.tsx:13]: added new line

File: pages/admin/users/[userToken].tsx [hunk 25]
```diff
@@ -43,40 +57,55 @@ export default function SingleUser({}: Props) {
...
-      </TabPanel>
-      <TabPanel value={selectedTabIndex} index={2}>
-        {user && <PeriodTracker id={user?.id} />}
-      </TabPanel>
-      <TabPanel value={selectedTabIndex} index={3}>
-        {user && <PrivateChat user={user} />}
-      </TabPanel>
+    <div>
+      <Navigation />
+
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/users/[userToken].tsx:62]: removed new line

File: pages/admin/videoProgress/index.tsx [hunk 26]
```diff
@@ -1,36 +1,69 @@
...
+  const { token } = useContext(UserPropsContext);
+
   const viewUser = (userToken: string) => {
     router.push(`/admin/users/[userToken]`, `/admin/users/${userToken}`);
   };
+
   const fetchData = async () => {
-    const data = await getVideoProgressOfAllUsers();
+    const data = await getVideoProgressOfAllUsers(allCourses);
+    // const data: any = [];
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/videoProgress/index.tsx:36]: removed the comment

File: pages/doctor/alldoctors/index.tsx [hunk 27]
```diff
@@ -1,9 +1,45 @@
...
+import { useRouter } from "next/router";
+import Swal from "sweetalert2";
 type Props = {};
 
 const index = (props: Props) => {
-  return <AllDoctors />;
+  const router = useRouter();
+  const [doctors, setDoctors] = useState<any[]>([]);
+  const { token } = useContext(UserPropsContext);
+  const fetchAllDoctors = async () => {
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/doctor/alldoctors/index.tsx:14]: added new line

File: pages/doctor/holidays/AllHolidays.tsx [hunk 28]
```diff
@@ -0,0 +1,226 @@
+import React, { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import BasicTable from "../../../src/components/Tables/BasicTable";
+import moment from "moment";
+import { UserPropsContext } from "../../../src/utils/context/user";
+import HolidayModal from "../../../src/components/holiday/HolidayModal";
+// import DownloadsComponent from "../../components/DownloadComponent";
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/doctor/holidays/AllHolidays.tsx:7]: removed

File: pages/doctor/holidays/index.tsx [hunk 29]
```diff
@@ -0,0 +1,69 @@
...
+import Doctorsidebar from "../../../src/components/Sidebar/Doctorsidebar";
+import Navigation from "../../../src/components/layout/Navigation";
+import Swal from "sweetalert2";
+type Props = {};
+
+const index = (props: Props) => {
+  const [open, setOpen] = useState(false);
+  const [holidays, setHolidays] = useState<any[]>([]);
+  const [holidaysCount, setHolidaysCount] = useState<number>(0);
+  // const { token } = React.useContext(UserPropsContext);
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/doctor/holidays/index.tsx:17]: removed comment

File: pages/doctor/holidays/index.tsx [hunk 30]
```diff
@@ -0,0 +1,69 @@
...
+  const fetchAllHolidays = async (doctorId: number) => {
+    const result = await getAllHolidays(doctorId);
+    if (result.isError) {
+      return;
+    }
+    setHolidays(result.rows);
+    setHolidaysCount(result.count);
+  };
+  let token: any = "";
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/doctor/holidays/index.tsx:36]: added new line.
The componet UI is slightly different than other.  Try to refactor it

File: pages/doctor/telehealth/DoctorBookings.tsx [hunk 31]
```diff
@@ -1,13 +1,23 @@
...
+} from "../../../src/utils/api/doctorUser";
+import {
+  getPrescriptionByUserId,
+  getPrescriptionByAppointmentId,
+} from "../../../src/utils/api/prescription";
+import { updateAttendanceOfDoctor } from "../../../src/utils/api/doctorUser";
+import DownloadsComponent from "../../../src/components/DownloadComponent";
+import PrescriptionModal from "./PrescriptionModal";
+import ShowPrescriptionModal from "./ShowPrescriptionModal";
+// import SwapSlotModal from "./SwapSlots";
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/doctor/telehealth/DoctorBookings.tsx:20]: removed comments

File: pages/doctor/telehealth/DoctorBookings.tsx
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/doctor/telehealth/DoctorBookings.tsx:1]: removed the unused code

File: pages/doctor/telehealth/MedicineSelector.tsx [hunk 32]
```diff
@@ -0,0 +1,379 @@
...
+    const fetchData = async () => {
+      const result = await getDoctorAvailabiltyForPrescription({
+        doctorId: appointment.doctorId,
+      });
+      setDoctorAvailability(result);
+    };
+    fetchData();
+  }, []);
+
+  console.log(doctorAvailability);
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/doctor/telehealth/MedicineSelector.tsx:74]: remove the logs

File: pages/doctor/telehealth/ShowPrescriptionModal.tsx [hunk 33]
```diff
@@ -0,0 +1,131 @@
...
+            open={show}
+            handleClose={handleClose}
+            user={user}
+            prescriptions={prescriptions?.data[prescriptionId[0]]}
+            doctor={doctor}
+            followup={prescriptionData?.followupDetails}
+            bookingDate={bookingDate}
+            // bookingdate={bookingdate}
+          />
+          {/* {Object.keys(prescriptionData).map(
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/doctor/telehealth/ShowPrescriptionModal.tsx:112]: removed the unused code

File: pages/doctor/users/Users.tsx [hunk 34]
```diff
@@ -10,6 +10,7 @@ import { getAllUsersByDoctor } from "../../../src/utils/api/doctorUser";
 // } from "../../../utils/api/admin/adminUser";
 // import DownloadsComponent from "../../DownloadComponent";
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/doctor/users/Users.tsx:11]: removed the comments

File: pages/doctor/users/index.tsx [hunk 35]
```diff
@@ -1,12 +1,36 @@
...
+        confirmButtonText: "Ok",
+      });
+      router.push("/staff/login");
+    }
+  }, []);
+
   return (
-    <div className="section-wrapper">
-      <Users key="allUsers" />
+    <div className="">
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/doctor/users/index.tsx:28]: removed classname

File: pages/profile/edit.tsx [hunk 36]
```diff
@@ -86,6 +97,10 @@ export const EditProfileForm = ({
         },
       });
 
+      if (data.otherProgram) {
+        data.program = data.otherProgram;
+      }
+      delete data.otherProgram;
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/profile/edit.tsx:103]: actually remaining keys are not part of the tracking variables that's why I didnt make it in one object

File: pages/staff/login/index.tsx (hunk 1 above)
Comment (yashnerkar) [reply pages/staff/login/index.tsx:112]: removed the comments

File: src/components/Admin/AppointmentTracker/AppointmentCard.tsx (hunk 2 above)
Comment (yashnerkar) [reply src/components/Admin/AppointmentTracker/AppointmentCard.tsx:4]: remvoed the comment

File: src/components/Admin/AppointmentTracker/AppointmentCard.tsx (hunk 3 above)
Comment (yashnerkar) [reply src/components/Admin/AppointmentTracker/AppointmentCard.tsx:127]: I have kept it for future reference. So not removing for now

File: src/components/Admin/CasePaperTracker/index.tsx
Comment (yashnerkar) [reply src/components/Admin/CasePaperTracker/index.tsx:1]: removed

File: src/components/Admin/PrescriptionTracker/PrescriptionCard.tsx (hunk 4 above)
Comment (yashnerkar) [reply src/components/Admin/PrescriptionTracker/PrescriptionCard.tsx:67]: this are actually the modal default styles that's why I have kept there!

File: src/components/DownloadComponent/index.tsx (hunk 6 above)
Comment (yashnerkar) [reply src/components/DownloadComponent/index.tsx:14]: removed

File: src/components/Prescription/PrescriptionTable.tsx (hunk 7 above)
Comment (yashnerkar) [reply src/components/Prescription/PrescriptionTable.tsx:20]: removed the comments

Comment (yashnerkar) [reply src/components/Prescription/PrescriptionTable.tsx:20]: removed the comments

File: src/components/Prescription/index.tsx (hunk 8 above)
Comment (yashnerkar) [reply src/components/Prescription/index.tsx:64]: removed the comments

File: src/components/holiday/HolidayModal.tsx
Comment (yashnerkar) [reply src/components/holiday/HolidayModal.tsx:1]: I have used this component in AllHoliday component in the doctor section as well as in admin section. used a single component in both the cases

File: src/components/logout/Logout.tsx (hunk 10 above)
Comment (yashnerkar) [reply src/components/logout/Logout.tsx:4]: removed the comments

File: src/theme/global.scss (hunk 13 above)
Comment (yashnerkar) [reply src/theme/global.scss:940]: resolved,added responsive of the navigationbar

File: src/theme/global.scss (hunk 11 above)
Comment (yashnerkar) [reply src/theme/global.scss:5]: removed the unused variable $md=768

File: src/utils/api/index.tsx
Comment (yashnerkar) [reply src/utils/api/index.tsx:1]: removed the comments

File: src/theme/global.scss (hunk 12 above)
Comment (yashnerkar) [reply src/theme/global.scss:1051]: resolved

File: pages/admin/holidays/index.tsx [hunk 37]
```diff
@@ -0,0 +1,73 @@
...
+    if (result.isError) {
+      return;
+    }
+    setHolidays(result.rows);
+    setHolidaysCount(result.count);
+  };
+  let token: any = "";
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/holidays/index.tsx:38]: added new line

File: pages/admin/holidays/index.tsx [hunk 38]
```diff
@@ -0,0 +1,73 @@
...
+      Swal.fire({
+        icon: "error",
+        title: "Oops...",
+        text: "You are not authorized to view this page",
+      });
+      router.push("/staff/login");
+    }
+  }, [token, open]);
+
+  console.log(holidays);
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/admin/holidays/index.tsx:54]: removed console log

File: pages/chats/myna.tsx [hunk 39]
```diff
@@ -36,6 +36,7 @@ function SingleChat({}: Props) {
               userName: userName,
               chatID: user?.userToken,
               userRole: "user",
+              receiverName: "Myna Expert",
```
[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/chats/myna.tsx:39]: changed the receiver number to env in all files

[the remark this answers is not in the document]
Comment (yashnerkar) [reply pages/chats/myna.tsx:39]: changed the receiver number to env in all files

File: src/components/DashboardCards/index.tsx (hunk 5 above)
Comment (yashnerkar) [reply src/components/DashboardCards/index.tsx:12]: added css

**Review by enroot-mumbai — APPROVED** [review]

**Pull request merged**